src/core/api/coap_api.cpp

```diff
@@ -35,135 +35,138 @@
 
 #include <openthread/coap.h>
 
-#include "coap/coap_header.hpp"
+#include "coap/coap_message.hpp"
 #include "common/instance.hpp"
 
 #if OPENTHREAD_ENABLE_APPLICATION_COAP
 
 using namespace ot;
 
-void otCoapHeaderInit(otCoapHeader *aHeader, otCoapType aType, otCoapCode aCode)
+otMessage *otCoapNewMessage(otInstance *aInstance, const otMessageSettings *aSettings)
 {
-    Coap::Header *header = static_cast<Coap::Header *>(aHeader);
-    header->Init(aType, aCode);
+    Message * message;
+    Instance &instance = *static_cast<Instance *>(aInstance);
+
+    if (aSettings != NULL)
+    {
+        VerifyOrExit(aSettings->mPriority <= OT_MESSAGE_PRIORITY_HIGH, message = NULL);
+    }
+
+    message = instance.GetApplicationCoap().NewMessage(aSettings);
+
+exit:
+    return message;
 }
 
-void otCoapHeaderSetToken(otCoapHeader *aHeader, const uint8_t *aToken, uint8_t aTokenLength)
+void otCoapMessageInit(otMessage *aMessage, otCoapType aType, otCoapCode aCode)
 {
-    static_cast<Coap::Header *>(aHeader)->SetToken(aToken, aTokenLength);
+    Coap::Message *message = static_cast<Coap::Message *>(aMessage);
+    message->Init(aType, aCode);
```

Comment:
nit: change to `static_cast<Coap::Message *>(aMessage)->Init(aType, aCode)` to be consistent with below?